tests/unit/multiclass/BaggingMachine_unittest.cc

```diff
@@ -151,7 +151,8 @@ TEST_F(BaggingMachine, classify_CART)
 	EXPECT_EQ(1.0,res_vector[4]);
 
 	auto eval = some<CMulticlassAccuracy>();
-	EXPECT_NEAR(0.642857,c->get_oob_error(eval),1e-6);
+	c->put("oob_error_metric", eval);
+	EXPECT_NEAR(0.642857,c->get<float64_t>("oob_error"),1e-6);
```

Comment:
not sure I like the put/get here. Maybe should have get_oob_error, and have another method that is exposed to the interface and added to parameter framework, i.e.
```cpp
watch_method("oob_error", &CBagginMachine::get_oob_error_interface)
...
float64_t CBagginMachine::get_oob_error_interface()
{
   require(m_oob_error_metric, "error message");
   return get_oob_error(m_oob_error_metric);
}
```

Reply:
would then also have the respective setter to be used in the C++ interface

Reply:
Not sure I understand the suggestion/problem Gil
This pattern is what we used for other similar situations, but the error check for the metric for sure needs to be done somewhere

Reply:
Btw related to the constexpr Viktor was asking about, any news in making the map compile time? :D

Reply:
@karlnapf the issue is that a user of the C++ API has to do a runtime lookup of the parameter map and then a any_cast..
There is a bit of progress with the compile time map. I have been thinking maybe we can just use an array and do a linear search. It would be easier to implement and a map might not be worth it for fewer than 20 or so key/value pairs

Reply:
We could do some stats on how many parameters we have on average ...

Reply:
Is the cast a problem if the method that is called is expensive (because it computes something)?

Reply:
I am not sure how much more expensive, but I would expect so. I mean you need to get rtti.. But I guess as long as you don't use it in a loop it's fine!